### PR TITLE
Remove babel and electron-webpack

### PR DESCRIPTION
**Pull request by jeffpeterson**
- :bulb: hot reload for react components and css! fairly snappy too
- :bulb: static files can be imported; default export is the url as a string
- ~:bug: gotta reload when first starting the app~
  - added a custom start script that makes sure everything is started in the right order
- ~:bug: bootup is maybe slower; fixable~
  - the start script only needs to load webpack once, so the overhead is reduced
- ~:bug: prod builds probably don't work yet; should just require a few tweaks~
  - electron-webpack had some built-in config that i had to extract

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -6,11 +6,13 @@
   "repository": "https://github.com/inkandswitch/pushpin",
   "scripts": {
     "build-deps": "electron-builder install-app-deps",
-    "compile": "electron-webpack",
-    "start": "electron-webpack dev",
+    "compile": "webpack",
```
Comment (mjtognetti) [inline package.json]: refreshing to see a good ol' `webpack`

File: src/renderer/index.tsx [hunk 2]
```diff
@@ -9,35 +9,19 @@ import raf from 'random-access-file'
 import DiscoverySwarm from 'discovery-cloud-client'
 
 import { HYPERMERGE_PATH, WORKSPACE_URL_PATH } from './constants'
+import Root from './components/Root'
```
Comment (mjtognetti) [inline src/renderer/index.tsx]: nice

File: webpack.config.ts [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+import HardSourcePlugin from 'hard-source-webpack-plugin'
+import ForkTsCheckerPlugin from 'fork-ts-checker-webpack-plugin'
+
+interface Options {
+  isDev: boolean
+}
+
+const cacheDirectory = undefined // path.resolve(__dirname, ".cache")
+
+const tsRule: webpack.Rule = {
```
Comment (mjtognetti) [inline webpack.config.ts]: this is cool, really nice way of structuring this. Always thought nested webpack config was so ugly.

File: src/renderer/ContentTypes.ts [hunk 4]
```diff
@@ -38,6 +38,12 @@ function register(contentType: ContentType) {
 
   log('register', entry)
 
+  if (registry[type]) {
+    // Allow re-registration to support HMR
+    // throw new Error(`Type already registered: ${type}`)
```
Comment (pvh) [inline src/renderer/ContentTypes.ts]: you can delete this commented-out line

File: src/renderer/components/workspace/TitleBar.css [hunk 5]
```diff
@@ -20,7 +20,7 @@
 }
 
 button[disabled].TitleBar__menuItem .fa {
-  color: #CCC;
+  color: #ccc;
```
Comment (pvh) [inline src/renderer/components/workspace/TitleBar.css]: Do you have different formatting settings from @mjtognetti ? Seems weird to see this here.

File: src/renderer/ibm-plex.css [hunk 6]
```diff
@@ -0,0 +1,64 @@
+@font-face {
+  font-family: 'IBM Plex Mono';
+  font-style: normal;
+  font-weight: 400;
+  src: local('IBM Plex Mono'), local('IBMPlexMono'),
+    url('../../node_modules/@ibm/plex/IBM-Plex-Mono/fonts/complete/woff2/IBMPlexMono-Regular.woff2')
```
Comment (pvh) [inline src/renderer/ibm-plex.css]: if we're going to do this, perhaps we should just properly vendor the font files and do away with the node module?

File: src/renderer/index.tsx [hunk 7]
```diff
@@ -90,11 +87,13 @@ function initWorkspace(repo: RepoFrontend) {
   }
 
   const workspace = <Root repo={repo} url={workspaceUrl} />
-  const element = document.getElementById('app')
+  const element = document.createElement('div')
+  element.id = 'app'
+  document.body.appendChild(element)
```
Comment (pvh) [inline src/renderer/index.tsx]: this is weird, what's going on here and is this the best way to do this?

**Comment by pvh** [comment]
Overall, I'm pretty happy with the code but something about the migration has changed the application directory name from Electron (presumably what webpack-electron sets) to PushPin. This means that without a migration we lose all of our users' data. Thoughts on what to do about that?

File: src/renderer/components/workspace/TitleBar.css (hunk 5 above)
Comment (jeffpeterson) [reply src/renderer/components/workspace/TitleBar.css]: Oh shoot, yeah, I think I have prettier running on css, json, etc.

File: src/renderer/index.tsx (hunk 7 above)
Comment (jeffpeterson) [reply src/renderer/index.tsx]: electron-webpack included an `#app` div in it's custom html template thing. Now, webpack is generating an `index.html` for us which doesn't include that div.

React can't mount directly onto `document.body`, so we just create the mounting element when the app boots. I think it's fairly standard.

**Comment by jeffpeterson** [comment]
I added user data migration. If you already ran the app on this branch without copying `{userData}/Electron/pushpin-v10` to `{userData}/pushpin/pushpin-v10` by hand, you'll need to delete or move the `{userData}/pushpin/pushpin-v10` directory.
Otherwise, if you haven't already ran the app on this branch, it should just work.

File: src/renderer/ibm-plex.css (hunk 6 above)
Comment (jeffpeterson) [reply src/renderer/ibm-plex.css]: added trello card and TODO comment to follow up with this

**Pull request merged**